### ISSUE TITLE
EnableLog_DisableLog

### ISSUE THREAD

**Issue by Paliha** [x64dbg/x64dbg#1416]
```
@log1:
EnableLog // if you put a breakpoint, is interrupted and pass press tab
log "kusd : {p:temp}" // then the log is saved
DisableLog
jmp @log1
```

if this is not done, then the log only this:
Logging will be enabled.
Logging will be disabled.

**Comment by mrexodia** [comment]
My guess is that there is no waiting before the command is processed in the GUI. Can be fixed with a `BridgeResult` instance.

**Comment by Paliha** [comment]
then not quite clear why in the scripting engine need these instructions, if the engine is not priority

**Comment by mrexodia** [comment]
The scripting engine is unrelated. Commands are used for more things than
that.

**Comment by Paliha** [comment]
Ок. !!!
In the tab "script", section "information",it is not used, why not make the logs for scripts ? It will be more convenient + very tabs appear next to the "Watch" and "Struct" to have everything before my eyes Because of the team "NSG" and "MSGYN" block switching between tabs. The team of "ASM" is the ability to format the string ?

**Comment by mrexodia** [comment]
Yeah that was the plan for the information column, but it was never added. I added the string formatting to `asm` if you have more commands please tell me.

**Comment by Paliha** [comment]
Thank you !
>if you have more commands please tell me.
if possible, string data in variables to use

**Comment by mrexodia** [comment]
String variables do actually exist, but all their functions are undocumented because it was never finished. It would be possible to add syntax to print the content of string variables with string formatting though...

**Comment by Paliha** [comment]
Ок. show you the problem on the example, and why the need for variables to string data.
instruction LEA RAX, [0x140000000](while the patterns for the 8 registers), is unique in its distance to the base image.There is no functionality to the script to find all. This is the beginning of handlers indices and distances to separate parts with instructions located in a chaotic manner, all the section of protection.

**Comment by mrexodia** [comment]
@Paliha good point actually that `rip+x` should be converted to something else...

**Comment by Paliha** [comment]
findasm/asmfind format the string ? ))

**Comment by mrexodia** [comment]
I added string formatting to the findasm command. As for the other part, I have absolutely no idea what you are talking about.

**Comment by Paliha** [comment]
"I added string formatting to the findasm command"
Super....)))
"As for the other part, I have absolutely no idea what you are talking about."
In the lower left corner of the widget "HexDump". In Ollydbg child Windows and get a window handle is not difficult. in your project I can't find where are bookmarks in Qtwidgets"HexDump" ..)) or you add another 2 tabs for script and log

**Comment by mrexodia** [comment]
There is no support for manipulating any of those windows, nor can you get
window handles for them. This is by design and if you want to change
something you can check out the source code and suggest changes...

**Comment by mrexodia** [comment]
> Oh Oh...it turns out that team findasm/asmfind and reffind/findref/ref with one functionality.
> int RefFind (...) {return refInfo.refcount;}
> Here it is necessary to find the first match, and return address.

I don't quite understand what you mean here... If you mean that scripts should be able to retrieve the address found, that might be a nice feature, but not part of the current behavior (see the documentation for that).

As for the rest, if you want something to change or you found a bug, please open **separate** issues. This way they can be classified and handled separately too. I have some on my personal list, but that list only gets longer over time so probably it will be forgotten.

**Comment by Paliha** [comment]
> I don't quite understand what you mean here.
https://exelab.ru/f/index.php?action=vthread&forum=13&topic=19719&page=34#7
1000 methods/functions for initialization transferred to the protection section, and they must return to their place. The pointer which moved there in the table init, and in sections of code written instruction unconditional branch
findasm "JMP {p:[table_init]}", code
Standard task functionality no. Even the address of the jump code are unable to

**Comment by mrexodia** [comment]
Okay I added some of the functionality you requested. 

- https://github.com/x64dbg/x64dbg/commit/8e601c72bfe37f9f29b29b5ee0877a420764a2a1
- https://github.com/x64dbg/x64dbg/commit/aaf610de547eb426db29a9d046f16166e257d288
- https://github.com/x64dbg/x64dbg/commit/62d3ac7d177581adc12cb6886b54d45303ebc3c7

**Comment by Paliha** [comment]
>8e601c7
>aaf610d
Super !!! )))
>62d3ac7
1.with the window reference is obtained for a long time
2.n each tab, 1 address, for a total of 1000 bookmarks !!!!
and it is possible to first match and no window links ?
nd in a variable address and not a number
if more than one, we simply change the beginning of the search.
findasm "JMP {p:current}", code
log "{p:$result}"
add $result},5

**Comment by mrexodia** [comment]
1. The `ref.addr` command can and will only work on the last reference view because of design limitations.
2. I opened issue #1494 for that. I only added `ref.addr` so you could accomplish what you wanted, but notice that using `findasm` for lots of matches is much faster than searching once every time...

I don't really understand what you mean with the "bookmarks" though. Do you just mean the entries in the reference view or are you referring to the bookmark system in x64dbg?

**Comment by Paliha** [comment]
Why continue to search if there is always one instruction ?
A waste of time
Why for each user to create reference dialog?
A waste of time
Find the first instruction, the return address.

**Comment by mrexodia** [comment]
Searching for all instructions is faster because you only have to read+allocate the complete memory page once :)
